spec/abstract/Decode_A.thy

```diff
@@ -532,30 +531,40 @@ where
         odE
       | _ \<Rightarrow> throwError (InvalidCapability 1);
       returnOk $ InvokeSchedContextUnbindObject sc_ptr cap
-    odE
-  | SchedContextUnbind \<Rightarrow> doE
-      cap \<leftarrow> returnOk $ hd excaps;
-      tcb_ptr \<leftarrow> returnOk $ obj_ref_of cap;
-      ct_ptr \<leftarrow> liftE $ gets cur_thread;
-      whenE (tcb_ptr = ct_ptr) $ throwError IllegalOperation;
-      returnOk $ InvokeSchedContextUnbind sc_ptr cap odE
-  | SchedContextYieldTo \<Rightarrow> doE
+    odE"
+
+definition
+  decode_sched_context_yield_to ::
+  "obj_ref \<Rightarrow> obj_ref option \<Rightarrow> (sched_context_invocation, 'z::state_ext) se_monad"
+where
+  "decode_sched_context_yield_to sc_ptr buffer \<equiv> doE
       sc \<leftarrow> liftE $ get_sched_context sc_ptr;
```

Comment:
Seeing that `sc` always appears with `sc_tcb` in this function, we can use `get_sc_obj_ref sc_tcb` here also.

Reply:
Would you like me to update this?

Reply:
That would be nice, I think. I hope it’s not too much more work.

Reply:
Well, it's in the abstract spec, so I'll have to check all layers and then test all layers again.

I did not modify this line originally. Are we wanting to do this for just this function, or all the functions in the abstract spec?

Reply:
I thought this is the function you added newly. Maybe you copied this bit from somewhere else? What I am talking about is what I can see inside this function. Not sure about others, but I don’t think it applies to others.

Reply:
Yeah this was all in one massive `decode_sched_context_invocation` function. 

OK

Reply:
I thought that, if you are going to move lemmas to Lib, you are likely to need to rebuild anyway.

Reply:
The lemma you mentioned moving to `Corres_UL` was preexisting. I don't want to move that

Reply:
Ah I see. Well, in that case, we should probably not rename that one but rename the other one.

Reply:
The preexisting one had the stupid name. It had maybe only one or maybe two uses in Refine. The lemma I added is named in analogy with `corres_split` and `corres_splits'`

Reply:
As we have seen repeatedly recently, we don’t do that kind of cleanup in MCS. We do it on master. You are welcome to do it on master. You can add your lemmas, too.